File: Backend/nuka-pos/src/main/java/com/nuka/nuka_pos/api/branch/Branch.java
package com.nuka.nuka_pos.api.branch;

import com.nuka.nuka_pos.api.currency.Currency;
import com.nuka.nuka_pos.api.organization.Organization;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


/**
 * Entity representing a Branch within an Organization.
 */
@Entity
@Table(name = "branches")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Branch {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String location;
    private String phone;
    private String email;

    private Boolean isHeadOffice;


    @ManyToOne
    @JoinColumn(name = "organization_id", nullable = false)
    private Organization organization;

    @ManyToOne
    @JoinColumn(name = "currency_id", nullable = false)
    private Currency currency;
}

File: Frontend/Nuka_POS/lib/models/branch.dart
class Branch {
  final int? id;
  final String name;
  final String location;
  final String phone;
  final String email;
  final bool isHeadOffice;
  final int organizationId;
  final int currencyId;

  Branch({
    this.id,
    required this.name,
    required this.location,
    required this.phone,
    required this.email,
    required this.isHeadOffice,
    required this.organizationId,
    required this.currencyId,
  });

  factory Branch.fromJson(Map<String, dynamic> json) {
    return Branch(
      id: json['id'],
      name: json['name'],
      location: json['location'],
      phone: json['phone'],
      email: json['email'],
      isHeadOffice: json['isHeadOffice'],
      organizationId: json['organizationId'],
      currencyId: json['currencyId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'location': location,
      'phone': phone,
      'email': email,
      'isHeadOffice': isHeadOffice,
      'organization': {'id':organizationId},
      'currency': {'id':currencyId},
    };
  }
}
